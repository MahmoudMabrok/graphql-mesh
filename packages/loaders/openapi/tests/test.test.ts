import { printSchemaWithDirectives } from '@graphql-tools/utils';
import loadGraphQLSchemaFromOpenAPI from '../src/index.js';
import { generateOperations } from '../../../cli/src/commands/generate-operations.js';

describe('Test', () => {
  it('should generate the correct schema', async () => {
    // Generate the schema
    const schema = await loadGraphQLSchemaFromOpenAPI('test', {
      source: './fixtures/test.json',
      cwd: __dirname,
    });
    expect(printSchemaWithDirectives(schema)).toMatchSnapshot();

    // Generate the operations
    const documents = generateOperations(schema, { selectionSetDepth: 3 });
    expect(documents).toBeDefined();
  });
});
