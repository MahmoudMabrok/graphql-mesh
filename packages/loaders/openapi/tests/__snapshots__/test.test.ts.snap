// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should generate the correct schema 1`] = `
"schema {
  query: Query
}

directive @oneOf on OBJECT | INTERFACE

directive @globalOptions(sourceName: String, endpoint: String, operationHeaders: ObjMap, queryStringOptions: ObjMap, queryParams: ObjMap) on OBJECT

directive @httpOperation(path: String, operationSpecificHeaders: ObjMap, httpMethod: HTTPMethod, isBinary: Boolean, requestBaseBody: ObjMap, queryParamArgMap: ObjMap, queryStringOptionsByParam: ObjMap) on FIELD_DEFINITION

type Query @globalOptions(sourceName: "test") {
  some_path: some_path_200_response @httpOperation(path: "/some/path", operationSpecificHeaders: "{\\"accept\\":\\"application/json\\"}", httpMethod: GET)
}

union some_path_200_response = components_BasicPropertyOption | components_ArrayPropertyOption

type components_BasicPropertyOption {
  commonProperty: String
}

type components_ArrayPropertyOption {
  commonProperty: [String]
}

scalar ObjMap

enum HTTPMethod {
  GET
  HEAD
  POST
  PUT
  DELETE
  CONNECT
  OPTIONS
  TRACE
  PATCH
}"
`;
